hack/test-integration.sh

```diff
@@ -96,16 +101,39 @@ loop="${TIMES:-1}"
 # hack/test-integration.sh Template*
 # hack/test-integration.sh "(WatchBuilds|Template)"
 tests=( $(go run "${OS_ROOT}/hack/listtests.go" -prefix="${OS_GO_PACKAGE}/${package}.Test" "${testexec}" | grep -E "${1-Test}") )
+
+if [[ -n "${junit_report}" ]]; then
+	tmp_dir="/tmp/openshift/test-integration"
+	mkdir -p ${tmp_dir}
+	test_output_file="${tmp_dir}/test-go.log"
+	test_error_file="${tmp_dir}/test-go-err.log"
+	junit_report_file="${tmp_dir}/report.xml"
+fi
 # run each test as its own process
 ret=0
+test_result="ok"
 pushd "${OS_ROOT}/${package}" &>/dev/null
+os::log::info "Running \`go test\`..."
+TESTSTARTTIME=$(date +%s%3N)
 for test in "${tests[@]}"; do
 	for((i=0;i<${loop};i+=1)); do
 		if ! (exectest "${test}" ${@:2}); then
 			ret=1
+			test_result="FAIL"
 		fi
 	done
 done
+TESTENDTIME=$(date +%s%3N)
+TESTDURATION=$(($TESTENDTIME - $TESTSTARTTIME))
+test_result_status="${test_result}        github.com/openshift/origin/test/integration    $(($TESTDURATION / 1000)).$(($TESTDURATION % 1000))s"
+if [[ -n "${test_output_file}" ]]; then
+	echo "${test_result_status}" | tee "${test_output_file}"
+	JUNIT_REPORT_OUTPUT="${test_output_file}" ARTIFACT_DIR="${tmp_dir}" os::test::junit::generate_report "gotest"
+else
+	echo "${test_result_status}"
```

Comment:
Not sure this is a super useful message either -- if you set `$JUNIT_REPORT_OUTPUT` in the `if [[ -n "${JUNIT_REPORT:-}" ]]` stanza like we do for other tests, and use `$ARTIFACT_DIR` from the setup scripts, this could look like:

```sh
echo "${test_result}        github.com/openshift/origin/test/integration    $((test_duration / 1000)).$((test_duration % 1000))s" >> "${JUNIT_REPORT_OUTPUT:-/dev/null}"
os::test::junit::generate_report "gotest"
```